src/main/resources/Untitled4.sql

```diff
@@ -0,0 +1,519 @@
+FROM
+    products   prod
+    INNER JOIN orders     ord ON prod.mfr_id = ord.mfr
+GROUP BY
+    mfr_id,
+    product_id,
+    description,
+    price,
+    qty_on_hand
+HAVING
+    SUM(qty) > ( 0.75 * qty_on_hand )
+ORDER BY
+    qty_on_hand DESC;
+
+/* 1. Скільки яких товірів купляли (скільки штук і чого купили всього) */
+
+SELECT
+    p.description,
+    o.product,
+    SUM(o.qty)
+FROM
+    products   p
+    INNER JOIN orders     o ON p.product_id = o.product
+GROUP BY
+    o.product,
+    p.description;
+
+/*2. Вивести товари, які купили більше 3-оьх разів (були в трьох і більше замовленнях) */
+
+SELECT
+    products.description,
+    COUNT(orders.product) AS count
+FROM
+    products
+    INNER JOIN orders ON products.product_id = orders.product
+GROUP BY
+    orders.product,
+    products.description
+HAVING
+    COUNT(orders.product) > 3;
+
+/*3. Вивести офіси в яких працює більше трьох менеджерів:*/
+
+SELECT
+    off.office,
+    off.city,
+    COUNT(s.rep_office)
+FROM
+    salesreps   s
+    INNER JOIN offices     off ON s.rep_office = off.office
+GROUP BY
+    off.office,
+    off.city
+HAVING
+    COUNT(s.rep_office) > 3;
+--приклад з підзапитом:
+
+SELECT
+    office,
+    city,
+    rep_office
+FROM
+    (
+        SELECT
+            offices.office,
+            offices.city,
+            COUNT(salesreps.rep_office) AS rep_office
+        FROM
+            offices
+            INNER JOIN salesreps ON offices.office = salesreps.rep_office
+        GROUP BY
+            offices.office,
+            offices.city
+    ) of_city_count
+WHERE
+    of_city_count.rep_office > 3;
+
+/*4. Вивести середній вік по офісу*/
+
+SELECT
+    o.office,
+    o.city,
+    AVG(s.age)
+FROM
+    salesreps   s
+    INNER JOIN offices     o ON s.rep_office = o.office
+GROUP BY
+    o.office,
+    o.city;
+
+/*5. Дати перелік виробників що виготовляють більше 4-ьох товарів  */
+
+SELECT
+    pr.mfr_id,
+    COUNT(pr.product_id)
+FROM
+    products pr
+GROUP BY
+    pr.mfr_id
+HAVING
+    COUNT(pr.product_id) > 4;
+
+/*6. Дати загальну кільсть товарів, що зберігаються на складі по виробникам що виготовляють більше 4-ьох товарів*/
+
+SELECT
+    pr.mfr_id,
+    COUNT(pr.product_id),
+    SUM(pr.qty_on_hand)
+FROM
+    products pr
+GROUP BY
+    pr.mfr_id
+HAVING
+    COUNT(pr.product_id) > 4;
+
+/* Варiант з пiдзапитом*/
+
+SELECT
+    pr.mfr_id,
+    COUNT(pr.product_id),
+    SUM(pr.qty_on_hand)
+FROM
+    products pr
+    INNER JOIN (
+        SELECT
+            mfr_id,
+            COUNT(product_id),
+            SUM(qty_on_hand)
+        FROM
+            products
+        GROUP BY
+            mfr_id
+        HAVING
+            COUNT(product_id) > 4
+    ) mfr ON pr.mfr_id = mfr.mfr_id
+GROUP BY
+    pr.mfr_id;
+
+/* 7. Вивести імя компанії, скільки вона тратила в середньому на покупки товарів виробників, котрі виготовляють більше 4-ьох товарів */
+
+SELECT
+    c.cust_num,
+    c.company,
+    AVG(s.sales) AS avg_sales,
+    o.mfr
+FROM
```

Comment:
s.sales this is a field which means all total sales, i think better to use AMOUNT from ORDERS